frontend/lib/models/filter.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+import 'artefact.dart';
+
+part 'filter.freezed.dart';
+
+@freezed
+class Filter with _$Filter {
```

Comment:
Just a small suggestion, maybe ArtefactFilter would be a better name here, as this filter only works for Artefacts.